diff --git a/credential.c b/credential.c
index 18098bd..f6212d4 100644
--- a/credential.c
+++ b/credential.c
@@ -71,6 +71,11 @@ static int credential_config_callback(const char *var, const char *value,
 			free(c->username);
 			c->username = xstrdup(value);
 		}
+	} else if (!strcmp(key, "password")) {
+		if (!c->password) {
+			free(c->password);
+			c->password = xstrdup(value);
+		}
 	}
 	else if (!strcmp(key, "usehttppath"))
 		c->use_http_path = git_config_bool(var, value);
