diff --git a/credential.c b/credential.c
index 18098bd..f6212d4 100644
--- a/credential.c
+++ b/credential.c
@@ -126,6 +126,11 @@ static int credential_config_callback(const char *var, const char *value,
 		if (!c->username_from_proto) {
 			free(c->username);
 			c->username = xstrdup(value);
+		}
+	} else if (!strcmp(key, "password")) {
+		if (!c->password) {
+			free(c->password);
+			c->password = xstrdup(value);
 		}
 	}
 	else if (!strcmp(key, "usehttppath"))
