diff --git a/lib/jekyll/commands/new.rb b/lib/jekyll/commands/new.rb
index 97549b1..73fd2da 100644
--- a/lib/jekyll/commands/new.rb
+++ b/lib/jekyll/commands/new.rb
@@ -94,16 +94,15 @@ platforms :mingw, :x64_mingw, :mswin, :jruby do
   gem "tzinfo-data"
 end
 
-# Performance-booster for watching directories on Windows
-gem "wdm", "~> 0.1.0", :install_if => Gem.win_platform?
-
 # kramdown v2 ships without the gfm parser by default. If you're using
 # kramdown v1, comment out this line.
-gem "kramdown-parser-gfm"
+gem "kramdown-parser-gfm", "~> 1.1"
 
 # Lock `http_parser.rb` gem to `v0.6.x` on JRuby builds since newer versions of the gem
 # do not have a Java counterpart.
 gem "http_parser.rb", "~> 0.6.0", :platforms => [:jruby]
+
+gem "webrick", "~> 1.8"
 RUBY
         end
 
@@ -159,7 +158,7 @@ RUBY
           Jekyll.logger.info "Running bundle install in #{path.cyan}..."
           Dir.chdir(path) do
             exe = Gem.bin_path("bundler", "bundle")
-            process, output = Jekyll::Utils::Exec.run("ruby", exe, "install")
+            process, output = Jekyll::Utils::Exec.run("@TERMUX_PREFIX@/bin/ruby", exe, "install", "--prefer-local")
 
             output.to_s.each_line do |line|
               Jekyll.logger.info("Bundler:".green, line.strip) unless line.to_s.empty?
