diff --git a/gems/bundled_gems b/gems/bundled_gems
index ae0a1ac..10eb213 100644
--- a/gems/bundled_gems
+++ b/gems/bundled_gems
@@ -21,3 +21,40 @@ rbs             3.4.0   https://github.com/ruby/rbs
 typeprof        0.21.9  https://github.com/ruby/typeprof
 debug           1.9.1   https://github.com/ruby/debug
 racc            1.7.3   https://github.com/ruby/racc
+json 2.7.2 https://github.com/flori/json
+csv 3.3.0 https://github.com/ruby/csv
+addressable 2.8.6 https://github.com/sporkmonger/addressable
+public_suffix 5.0.5 https://github.com/weppos/publicsuffix-ruby
+colorator 1.1.0 https://github.com/octopress/colorator
+em-websocket 0.5.3 http://github.com/igrigorik/em-websocket
+http_parser.rb 0.8.0 https://github.com/tmm1/http_parser.rb
+i18n 1.14.5 https://github.com/ruby-i18n/i18n
+concurrent-ruby 1.3.1 https://github.com/ruby-concurrency/concurrent-ruby
+jekyll-sass-converter 3.0.0 https://github.com/jekyll/jekyll-sass-converter
+sass-embedded 1.77.4 https://github.com/sass-contrib/sass-embedded-host-ruby
+google-protobuf 4.27.0 https://github.com/protocolbuffers/protobuf
+bigdecimal 3.1.8 https://github.com/ruby/bigdecimal
+jekyll-watch 2.2.1 https://github.com/jekyll/jekyll-watch
+listen 3.9.0 https://github.com/guard/listen
+rb-fsevent 0.11.2 https://github.com/thibaudgg/rb-fsevent
+rb-inotify 0.11.1 https://github.com/guard/rb-inotify
+ffi 1.16.3 https://github.com/ffi/ffi/wiki
+kramdown 2.4.0 https://github.com/gettalong/kramdown
+strscan 3.1.0 https://github.com/ruby/strscan
+kramdown-parser-gfm 1.1.0 https://github.com/kramdown/parser-gfm
+kramdown 2.4.0 https://github.com/gettalong/kramdown
+rexml 3.2.8 https://github.com/ruby/rexml
+strscan 3.1.0 https://github.com/ruby/strscan
+liquid 4.0.4 https://github.com/Shopify/liquid
+mercenary 0.4.0 https://github.com/jekyll/mercenary
+pathutil 0.16.2 http://github.com/envygeeks/pathutil
+forwardable-extended 2.6.0 http://github.com/envygeeks/forwardable-extended
+rouge 4.2.1 https://github.com/rouge-ruby/rouge
+safe_yaml 1.0.5 https://github.com/dtao/safe_yaml
+terminal-table 3.0.2 https://github.com/tj/terminal-table
+unicode-display_width 2.5.0 https://github.com/janlelis/unicode-display_width
+webrick 1.8.1 https://github.com/ruby/webrick
+ffi 1.17.0 https://github.com/ffi/ffi
+mini_portile2 2.8.7 https://github.com/flavorjones/mini_portile
+jekyll-feed 0.17.0 https://github.com/jekyll/jekyll-feed
+jekyll-seo-tag 2.8.0 https://github.com/jekyll/jekyll-seo-tag
