diff --git a/lib/rubygems/core_ext/kernel_gem.rb b/lib/rubygems/core_ext/kernel_gem.rb
index 4e09b95..aba8e25 100644
--- a/lib/rubygems/core_ext/kernel_gem.rb
+++ b/lib/rubygems/core_ext/kernel_gem.rb
@@ -33,6 +33,8 @@ module Kernel
   #   GEM_SKIP=libA:libB ruby -I../libA -I../libB ./mycode.rb
 
   def gem(gem_name, *requirements) # :doc:
+    $LOAD_PATH << "/home/builder/.termux-build/ruby/host-build/ruby-host/lib/ruby/gems/3.3.0/gems/mini_portile2-2.8.7/lib"
+    puts "LOAD_PATH: #{$LOAD_PATH}\n"
     skip_list = (ENV["GEM_SKIP"] || "").split(/:/)
     raise Gem::LoadError, "skipping #{gem_name}" if skip_list.include? gem_name
 
