diff --git a/lib/rubygems/core_ext/kernel_gem.rb b/lib/rubygems/core_ext/kernel_gem.rb
index 4e09b95..9e0c2f2 100644
--- a/lib/rubygems/core_ext/kernel_gem.rb
+++ b/lib/rubygems/core_ext/kernel_gem.rb
@@ -33,6 +33,10 @@ module Kernel
   #   GEM_SKIP=libA:libB ruby -I../libA -I../libB ./mycode.rb
 
   def gem(gem_name, *requirements) # :doc:
+    Gem.paths = {
+      'GEM_PATH' => "/home/builder/.local/share/gem/ruby/3.3.0:/home/builder/.termux-build/ruby/build/ruby/gems/3.3.0:/home/builder/.termux-build/ruby/host-build/ruby-host/lib/ruby/gems/3.3.0"
+    }
+    $LOAD_PATH << "/home/builder/.termux-build/ruby/host-build/ruby-host/lib/ruby/gems/3.3.0"
     skip_list = (ENV["GEM_SKIP"] || "").split(/:/)
     raise Gem::LoadError, "skipping #{gem_name}" if skip_list.include? gem_name
 
