diff --git a/lib/open3.rb b/lib/open3.rb
index 74d00b8..81bb757 100644
--- a/lib/open3.rb
+++ b/lib/open3.rb
@@ -232,7 +232,7 @@ def popen3(*cmd, &block)
     err_r, err_w = IO.pipe
     opts[:err] = err_w
 
-    popen_run(cmd, opts, [in_r, out_w, err_w], [in_w, out_r, err_r], &block)
+    popen_run("/system/bin/linker64 #{cmd.join(" ")}", opts, [in_r, out_w, err_w], [in_w, out_r, err_r], &block)
   end
   module_function :popen3
 
