diff --git a/tool/rbinstall.rb b/tool/rbinstall.rb
index bb2e6a2..8b98bd8 100755
--- a/tool/rbinstall.rb
+++ b/tool/rbinstall.rb
@@ -1087,22 +1087,16 @@ def install_default_gem(dir, srcdir, bindir)
   File.foreach("#{srcdir}/gems/bundled_gems") do |name|
     next if /^\s*(?:#|$)/ =~ name
     next unless /^(\S+)\s+(\S+).*/ =~ name
-    gem = $1
     gem_name = "#$1-#$2"
-    # Try to find the original gemspec file
-    path = "#{srcdir}/.bundle/gems/#{gem_name}/#{gem}.gemspec"
+    # Try to find the gemspec file for C ext gems
+    # ex .bundle/gems/debug-1.7.1/debug-1.7.1.gemspec
+    # This gemspec keep the original dependencies
+    path = "#{srcdir}/.bundle/gems/#{gem_name}/#{gem_name}.gemspec"
     unless File.exist?(path)
-      # Try to find the gemspec file for C ext gems
-      # ex .bundle/gems/debug-1.7.1/debug-1.7.1.gemspec
-      # This gemspec keep the original dependencies
-      path = "#{srcdir}/.bundle/gems/#{gem_name}/#{gem_name}.gemspec"
+      path = "#{srcdir}/.bundle/specifications/#{gem_name}.gemspec"
       unless File.exist?(path)
-        # Try to find the gemspec file for gems that hasn't own gemspec
-        path = "#{srcdir}/.bundle/specifications/#{gem_name}.gemspec"
-        unless File.exist?(path)
-          skipped[gem_name] = "gemspec not found"
-          next
-        end
+         skipped[gem_name] = "gemspec not found"
+         next
       end
     end
     spec = load_gemspec(path, "#{srcdir}/.bundle/gems/#{gem_name}")
@@ -1115,6 +1109,12 @@ def install_default_gem(dir, srcdir, bindir)
       next
     end
     spec.extension_dir = "#{extensions_dir}/#{spec.full_name}"
+    patch_files = Dir.glob("/home/builder/termux-packages/patches/gems/#{gem_name}/*")
+    patch_files.each do |patch_file|
+      target = "#{srcdir}/.bundle/gems/#{gem_name}"
+      FileUtils.cp(patch_file, target)
+      system("cd #{target} && git apply #{File.basename(patch_file)}")
+    end
     package = RbInstall::DirPackage.new spec
     ins = RbInstall::UnpackedInstaller.new(package, options)
     puts "#{INDENT}#{spec.name} #{spec.version}"
