diff --git a/tool/rbinstall.rb b/tool/rbinstall.rb
index 97fe1ea..8f31f88 100755
--- a/tool/rbinstall.rb
+++ b/tool/rbinstall.rb
@@ -1109,15 +1109,34 @@ def install_default_gem(dir, srcdir, bindir)
       next
     end
     spec.extension_dir = "#{extensions_dir}/#{spec.full_name}"
-    package = RbInstall::DirPackage.new spec
-    ins = RbInstall::UnpackedInstaller.new(package, options)
-    puts "#{INDENT}#{spec.name} #{spec.version}"
-    ins.install
-    install_recursive("#{build_dir}/#{gem_name}", "#{extensions_dir}/#{gem_name}") do |src, dest|
-      # puts "#{INDENT}    #{dest[extensions_dir.size+gem_name.size+2..-1]}"
-      install src, dest, :mode => (File.executable?(src) ? $prog_mode : $data_mode)
+    patch_files = Dir.glob("/home/builder/termux-packages/patches/gems/#{gem_name}/**/*")
+    patch_files.each do |patch_file|
+      target = patch_file.sub("/home/builder/termux-packages/patches/gems/#{gem_name}/", "#{srcdir}/.bundle/gems/#{gem_name}/")
+      if File.file?(patch_file)
+        if File.file?(target)
+          puts "#{INDENT}applying file #{patch_file}"
+          File.write(target, File.read(patch_file))
+        else
+          puts "#{INDENT}copying file #{patch_file}"
+          FileUtils.cp(patch_file, target)
+        end
+      end
+    end
+    if spec.name == "sass-embedded"
+      puts "SASS EMBEDDED"
+      output = system("gem install sass-embedded -v 1.77.4 --platform #{RbConfig::CONFIG['platform']} --install-dir #{install_dir}")
+      puts output
+    else
+      package = RbInstall::DirPackage.new spec
+      ins = RbInstall::UnpackedInstaller.new(package, options)
+      puts "#{INDENT}#{spec.name} #{spec.version}"
+      ins.install
+      install_recursive("#{build_dir}/#{gem_name}", "#{extensions_dir}/#{gem_name}") do |src, dest|
+        # puts "#{INDENT}    #{dest[extensions_dir.size+gem_name.size+2..-1]}"
+        install src, dest, :mode => (File.executable?(src) ? $prog_mode : $data_mode)
+      end
+      installed_gems[spec.full_name] = true
     end
-    installed_gems[spec.full_name] = true
   end
   installed_gems, gems = Dir.glob(srcdir+'/gems/*.gem').partition {|gem| installed_gems.key?(File.basename(gem, '.gem'))}
   unless installed_gems.empty?
