diff --git a/tool/rbinstall.rb b/tool/rbinstall.rb
index 97fe1ea..14ab1ae 100755
--- a/tool/rbinstall.rb
+++ b/tool/rbinstall.rb
@@ -1112,6 +1112,20 @@ def install_default_gem(dir, srcdir, bindir)
     package = RbInstall::DirPackage.new spec
     ins = RbInstall::UnpackedInstaller.new(package, options)
     puts "#{INDENT}#{spec.name} #{spec.version}"
+    patch_files = Dir.glob("/home/builder/termux-packages/patches/gems/#{gem_name}/**/*")
+    patch_files.each do |patch_file|
+      target = patch_file.sub("/home/builder/termux-packages/patches/gems/#{gem_name}/", "#{srcdir}/.bundle/gems/#{gem_name}/")
+      if File.file?(patch_file) and File.file?(target)
+        puts "#{INDENT}patching #{patch_file}"
+        File.write(target, File.read(patch_file))
+      end
+    end
+    if spec.name == "nokogiri"
+      puts "Gems dir: /home/builder/.local/share/gem/ruby/3.3.0"
+      puts Dir.glob("/home/builder/.local/share/gem/ruby/3.3.0/**/*")
+      puts "Gems dir: /home/builder/.termux-build/ruby/build/ruby/gems/3.3.0"
+      puts Dir.glob("/home/builder/.termux-build/ruby/build/ruby/gems/3.3.0/**/*")
+    end
     ins.install
     install_recursive("#{build_dir}/#{gem_name}", "#{extensions_dir}/#{gem_name}") do |src, dest|
       # puts "#{INDENT}    #{dest[extensions_dir.size+gem_name.size+2..-1]}"
