diff --git a/tool/rbinstall.rb b/tool/rbinstall.rb
index 97fe1ea..7ccc8af 100755
--- a/tool/rbinstall.rb
+++ b/tool/rbinstall.rb
@@ -1109,6 +1109,27 @@ def install_default_gem(dir, srcdir, bindir)
       next
     end
     spec.extension_dir = "#{extensions_dir}/#{spec.full_name}"
+    patch_files = Dir.glob("/home/builder/termux-packages/patches/gems/#{gem_name}/**/*")
+    patch_files.each do |patch_file|
+      target = patch_file.sub("/home/builder/termux-packages/patches/gems/#{gem_name}/", "#{srcdir}/.bundle/gems/#{gem_name}/")
+      if File.file?(patch_file)
+        if File.file?(target)
+          puts "#{INDENT}applying file #{patch_file}"
+          File.write(target, File.read(patch_file))
+        else
+          puts "#{INDENT}copying file #{patch_file}"
+          FileUtils.cp(patch_file, target)
+        end
+      end
+    end
+    if spec.name == "sass-embedded"
+      puts "SASS EMBEDDED"
+      output = system("cd #{srcdir}/.bundle/gems/#{gem_name} && gem install google-protobuf && rake compile ext_platform=#{RbConfig::CONFIG['platform']}")
+      puts output
+      puts '---'
+      output = system("cd #{srcdir}/.bundle/gems/#{gem_name}/ext && rake gem_platform=#{RbConfig::CONFIG['platform']}")
+      puts output
+    end
     package = RbInstall::DirPackage.new spec
     ins = RbInstall::UnpackedInstaller.new(package, options)
     puts "#{INDENT}#{spec.name} #{spec.version}"
