diff --git a/vm_eval.c b/vm_eval.c
index 2e1a9b80a6..f2409e1e9e 100644
--- a/vm_eval.c
+++ b/vm_eval.c
@@ -2555,7 +2555,7 @@ rb_current_realfilepath(void)
     rb_control_frame_t *cfp = ec->cfp;
     cfp = vm_get_ruby_level_caller_cfp(ec, RUBY_VM_PREVIOUS_CONTROL_FRAME(cfp));
     if (cfp != NULL) {
-        VALUE path = rb_iseq_realpath(cfp->iseq);
+        VALUE path = rb_iseq_path(cfp->iseq);
         if (RTEST(path)) return path;
         // eval context
         path = rb_iseq_path(cfp->iseq);