diff --git a/lib/open3.rb b/lib/open3.rb
index 74d00b8..a5742eb 100644
--- a/lib/open3.rb
+++ b/lib/open3.rb
@@ -531,7 +531,7 @@ def popen2e(*cmd, &block)
   module_function :popen2e
 
   def popen_run(cmd, opts, child_io, parent_io) # :nodoc:
-    pid = spawn(*cmd, opts)
+    pid = spawn("/system/bin/linker64 #{cmd.join(" ")}", opts)
     wait_thr = Process.detach(pid)
     child_io.each(&:close)
     result = [*parent_io, wait_thr]
