diff --git a/clang/lib/Driver/Driver.cpp b/clang/lib/Driver/Driver.cpp
index 93cddf742..bd27158ae 100644
--- a/clang/lib/Driver/Driver.cpp
+++ b/clang/lib/Driver/Driver.cpp
@@ -168,7 +168,8 @@ std::string Driver::GetResourcesPath(StringRef BinaryPath,
   // exact same string ("a/../b/" and "b/" get different hashes, for example).
 
   // Dir is bin/ or lib/, depending on where BinaryPath is.
-  std::string Dir = std::string(llvm::sys::path::parent_path(BinaryPath));
+  // std::string Dir = std::string(llvm::sys::path::parent_path(BinaryPath));
+  std::string Dir = "/data/data/sh.gourav.jekyllex/files/usr";
 
   SmallString<128> P(Dir);
   if (CustomResourceDir != "") {
@@ -179,7 +180,7 @@ std::string Driver::GetResourcesPath(StringRef BinaryPath,
     // With a static-library build of libclang, LibClangPath will contain the
     // path of the embedding binary, which for LLVM binaries will be in bin/.
     // ../lib gets us to lib/ in both cases.
-    P = llvm::sys::path::parent_path(Dir);
+    // P = llvm::sys::path::parent_path(Dir);
     // This search path is also created in the COFF driver of lld, so any
     // changes here also needs to happen in lld/COFF/Driver.cpp
     llvm::sys::path::append(P, CLANG_INSTALL_LIBDIR_BASENAME, "clang",
diff --git a/lld/COFF/Driver.cpp b/lld/COFF/Driver.cpp
index 22ee2f133..4a8af2d1e 100644
--- a/lld/COFF/Driver.cpp
+++ b/lld/COFF/Driver.cpp
@@ -645,7 +645,8 @@ void LinkerDriver::addClangLibSearchPaths(const std::string &argv0) {
   std::string lldBinary = sys::fs::getMainExecutable(argv0.c_str(), nullptr);
   SmallString<128> binDir(lldBinary);
   sys::path::remove_filename(binDir);                 // remove lld-link.exe
-  StringRef rootDir = sys::path::parent_path(binDir); // remove 'bin'
+  // StringRef rootDir = sys::path::parent_path(binDir); // remove 'bin'
+  StringRef rootDir = "/data/data/sh.gourav.jekyllex/files/usr";
 
   SmallString<128> libDir(rootDir);
   sys::path::append(libDir, "lib");
